torch/nn/parallel/distributed.py

```diff
@@ -1355,8 +1400,11 @@ def _inside_ddp_forward(self):
             DistributedDataParallel._active_ddp_module = None
 
     def _run_ddp_forward(self, *inputs, **kwargs):
-        with self._inside_ddp_forward():
+        if self._ddp_python_hook:
```

Comment:
might be clearer if this were named 'compiled_ddp_enabled' or if you worry about confusing between current compiled graph-split ddp, 'compiled_ddp_v2' or something

Reply:
I changed it to `_use_python_reducer`. This should be more explicit and the confusion between the original ddp compilation and the new one.